.github/workflows/docker.yml

```diff
@@ -0,0 +1,148 @@
+name: docker
+on:
+  push:
+  # TODO: Build on main?
+    branches:
+      - main
+    tags:
+      - '**'
+  # TODO: Remove pull_request before merging
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  # This job builds the binaries and uploads it as github artifacts.
+  # This allows us to use the same binaries for any release CI jobs.
+  build:
+    name: Build all linux architectures
+    runs-on: ubuntu-latest
+    steps:
+      - name: setup go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - uses: actions/checkout@v3
+        with:
+          # Fetch all tags
+          fetch-depth: 0
+
+      - name: Build on all supported architectures
+        run: |
+          set -e
+          ./scripts/release.sh
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: binaries-${{ github.sha }}
+          path: |
+            release*/*
+
+  # This job downloads the binaries previously uploaded as artifacts, and builds multi-arch images
+  build-docker-image:
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # Fetch all tags
+          fetch-depth: 0
+
+      - name: Prepare
+        id: prep
+        run: |
+          set -e
+          TAG=$( git describe --tags --dirty ) # E.g. v1.2.0-23-g60ee190
+          echo "TAG=$TAG" >> $GITHUB_ENV
+
+      # This step generates the docker tags
+      # type=ref,event=branch generates tag(s) on branch only
+      # type=semver generates tag(s) on tags only
+      # TODO: Remove `type=ref,event=pr` before merging PR
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            ${{ github.repository }}
+            ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=pr
+            # On main. E.g. main-abc01234
+            type=ref,suffix=-{{sha}},event=branch
+            # On tags. E.g. 'v0', 'v0.0', 'v0.0.0', and 'latest'
+            type=semver,pattern=v{{major}}
+            type=semver,pattern=v{{major}}.{{minor}}
+            type=semver,pattern=v{{major}}.{{minor}}.{{patch}}
```

Comment:
This should work on `main` (see [here](https://github.com/leojonathanoh/test-github-ci/actions/runs/4323587473/jobs/7547363641#step:3:64)):
```
    "tags": [
      "containernetworking/plugins:main-c3f69ea"
      "ghcr.io/containernetworking/plugins:main-c3f69ea"
    ],
```

and on `tags` (see [here](https://github.com/leojonathanoh/test-github-ci/actions/runs/4323587834/jobs/7547364376#step:3:70)):

```
containernetworking/plugins:v0
containernetworking/plugins:v0.6
containernetworking/plugins:v0.6.0
containernetworking/plugins:latest
ghcr.io/containernetworking/plugins:v0
ghcr.io/containernetworking/plugins:v0.6
ghcr.io/containernetworking/plugins:v0.6.0
ghcr.io/containernetworking/plugins:latest
```